controllers/volumereplicationgroup_controller.go

```diff
@@ -1648,49 +1648,84 @@ func (v *VRGInstance) validateVRStatus(volRep *volrep.VolumeReplication, state r
 		action = "demoted"
 	}
 
-	volRepCondition := findCondition(volRep.Status.Conditions, volrepController.ConditionCompleted)
-	if volRepCondition == nil {
-		v.log.Info("failed to get the Completed condition from VolumeReplication resource", "volRep",
-			volRep.Name)
+	// it should be completed
+	conditionMet, msg := isVRConditionMet(volRep, volrepController.ConditionCompleted, metav1.ConditionTrue)
+	if !conditionMet {
+		v.updateProtectedPVCConditionHelper(volRep.Name, PVCError, msg,
+			fmt.Sprintf("VolumeReplication resource for pvc not %s to %s", action, stateString))
 
-		msg := "Failed to get the Completed condition from VolumeReplication resource for pvc."
-		v.updateProtectedPVCCondition(volRep.Name, PVCError, msg)
+		return
+	}
 
-		return nil
+	// if primary, all checks are completed
+	if state == ramendrv1alpha1.Primary {
+		msg = "VolumeReplication resource for the pvc is replicating"
+		v.updateProtectedPVCCondition(volRep.Name, PVCReplicating, msg)
+
+		return
 	}
 
-	if volRepCondition.Status == metav1.ConditionFalse {
-		v.log.Info(fmt.Sprintf("VolumeReplication resource is not %s (%s)", action, volRep.Name))
+	// it should be resyncing, if secondary
+	conditionMet, msg = isVRConditionMet(volRep, volrepController.ConditionResyncing, metav1.ConditionTrue)
+	if !conditionMet {
+		v.updateProtectedPVCConditionHelper(volRep.Name, PVCError, msg,
+			"VolumeReplication resource for pvc not resyncing as Secondary")
 
-		msg := fmt.Sprintf("VolumeReplication resource for pvc not %s to %s", action, stateString)
-		v.updateProtectedPVCCondition(volRep.Name, PVCError, msg)
+		return
+	}
 
-		return nil
+	// it should not be degraded, if secondary
+	/* TODO: This needs a fix for https://github.com/csi-addons/volume-replication-operator/issues/101 based on which
```

Comment:
@raghavendrabhat commented this condition out till we get a fix for the other issue in VolRep. Without this VRG will never report a healthy secondary, and now with just the resync we avoid split brains and hence looks fine.